### ISSUE TITLE
Support for STS assumeRole

### ISSUE THREAD

**Issue by patkar** [test-kitchen/kitchen-ec2#344]
In our company we're solely using only one credentials set for a master account, were we can assumeRole into any other AWS account we manage. Would be good to have an assume_role parameter (with role ARN), so this driver will first obtain a new access credentials set from STS for the assumed role.

**Comment by patkar** [comment]
Additionaly it could extend to support the `role_arn` from `~/.aws/config`:

```
[profile client-project]
role_arn = arn:aws:iam::1122234567890:role/Developer
source_profile = pixelart
```

**Comment by rhyas** [comment]
@patkar Check out the PR I just submitted. We're using that internally to do exactly what this ticket is asking for, making use of the config file for source_profile and role_arn. All you need in the kitchen.yml is the name of the profile you're aiming to use, the SDK handles the rest.

**Comment by patkar** [comment]
Ok, your PR means, it is necessary to correctly configure every profile we want to assume in `~/.aws/config`. It's ok, at least something which forces us to manage that cleanly ^^

**Comment by rhyas** [comment]
Yes, previously the config file wasn't really used. Now it is, so if it's being used, it has to be formatted right. However, you still have all the other options as well, such as only using the credentials file, or using machine roles, environment variables, etc. This PR just moves all that detection back to the SDK and enables support for modes the SDK supports but the module didn't because it did discovery on it's own.

**Comment by patkar** [comment]
Yeah, forcing users to configure `~/.aws/config` correctly is a legit approach. IMHO every other application can profit from it too. Although I still see many not supporting it :)

Finally it's a better approach, then having a role_arn parameter and handling it itself. (Side note: Terraform AWS provider supports roles from `~/.aws/config` in the newest release :grin: Reason why I approached a `role_arn` param first, 'cause I was used to it through Terraform ^^).

**Comment by cheeseplus** [comment]
PR merged and 2.0.0 released, PR for ChefDK is incoming so this should be available in a `current` channel build soon.
